org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/matching/JavaSearchNameEnvironment.java

```diff
@@ -563,4 +573,18 @@ private static boolean isComplianceJava9OrHigher(IJavaProject javaProject) {
 	}
 	return CompilerOptions.versionToJdkLevel(javaProject.getOption(JavaCore.COMPILER_COMPLIANCE, true)) >= ClassFileConstants.JDK9;
 }
+
+private static boolean isOnModulePath(PackageFragmentRoot root) throws JavaModelException {
+	IClasspathEntry classpathEntry = root.getRawClasspathEntry();
+	IClasspathAttribute[] attributes = classpathEntry.getExtraAttributes();
```

Comment:
I see similar code in `org.eclipse.jdt.internal.core.ClasspathEntry.isModular()` and `org.eclipse.jdt.launching.JavaRuntime.isModule(IClasspathEntry, IJavaProject)`.

Looks like this should be just a default method on `IClasspathEntry`

Reply:
If we want to introduce new API, best to open an extra issue for that. Can you check the current change? Is that enough? If so, should I adjust also `JavaRuntime` (that seems to be in another plug-in) to use the new method? Though it might be accessing internal code then.

Reply:
Right, let split that in two changes.